templates/ehub.json

```diff
@@ -247,11 +263,25 @@
                         "APP_STATS_QUEUE_NAME": "[variables('statsQueueName')]",
                         "APP_PRINCIPAL_ID": "[reference(resourceId('Microsoft.Web/Sites', parameters('Application Name')), '2016-08-01', 'Full').identity.principalId]",
                         "APP_FILTER_JSON": "[variables('filterJson')]",
-                        "APP_FILTER_REGEX": "[variables('filterRegex')]"
+                        "APP_FILTER_REGEX": "[variables('filterRegex')]",
+                        "APPINSIGHTS_INSTRUMENTATIONKEY": "[if(variables('isApplicationInsightEnabled'),reference(concat('microsoft.insights/components/', variables('applicationInsightsName'))).InstrumentationKey,json('null'))]",
+                        "ApplicationInsightsAgent_EXTENSION_VERSION": "~2"
                     }
                 }
             ]
         },
+        {
+            "condition": "[variables('isApplicationInsightEnabled')]",
+            "type": "Microsoft.Insights/components",
+            "apiVersion": "2020-02-02",
+            "name": "[variables('applicationInsightsName')]",
+            "location": "[variables('location')]",
+            "kind": "web",
+            "properties": {
```

Comment:
I think we need to create Insights resource after function web app is created. Basically add the following.
```
"dependsOn": [
                        "[resourceId('Microsoft.Web/Sites', parameters('Application Name'))]"
                    ],
```

Could you please double-check